ESP8266/ESP8266.cpp

```diff
@@ -16,14 +16,22 @@
 
 #include "ESP8266.h"
 
+#define ATPARSER_BUFSIZE    256  //bytes
+#define ATPARSER_TIMEOUT    8000 //ms
+#define BAUDRATE            115200
+
 ESP8266::ESP8266(PinName tx, PinName rx, bool debug)
-    : _serial(tx, rx, 1024), _parser(_serial)
-    , _packets(0), _packets_end(&_packets)
+    :  _packets(0), _packets_end(&_packets)
 {
-    _serial.baud(115200);
-    _parser.debugOn(debug);
+    _fh = new BufferedSerial(tx, rx, BAUDRATE);
+    _parser =  new ATParser(*_fh, ATPARSER_BUFSIZE, ATPARSER_TIMEOUT, debug);
```

Comment:
Why are these dynamically allocated now? They can just be declared in the class, virtual inheritance will still work if passed by address.